### DISCUSSION TITLE
Export from Zim (or Zim's linked git repo) to Mediawiki

### DISCUSSION THREAD

**Discussion by bblackmoor** [zim-desktop-wiki/zim-desktop-wiki#2240] [Q&A] [answered]
I run a handful of Mediawiki sites. It works well and I am familiar with it, but I was looking for a better way to work on things offline. I found Zim.

Is there a known automated (or mostly-automated) workflow where one can edit a page in Zim, and import that wikitext to Mediawiki? I am comfortable with git, etc.

Thank you.

**Comment by introt** [comment, marked as answer]
While Zim's format is a kind of wiki syntax, you're generally going to have a bad time using a separate wiki instance to write content for another one due to non-syntax related things like in-wiki links etc.

If you're a masochist willing to risk death by a thousand cuts, you can either try (modifying and/or rewriting) [this old script](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/contrib/zim2trac.py), or exporting from Zim to MarkDown and using [Pandoc](https://pandoc.org/index.html) to convert the output into the MediaWiki format. Both approaches can be automated.

**Comment by bblackmoor** [comment]
I see. Thanks.

**Reply by introt** [reply]
You're welcome. If you're the only editor of your sites, you can use Zim to handle all content creation and export to static HTML pages, negating the need for server hosting. If you have multiple editors, you might want to also keep in mind that you're always running the risk of conflicts when editing offline, regardless of the software you use.

**Comment by lorenrosen** [comment]
Also, note there's a version control plugin that might help integrate with git.
